sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -2061,17 +2043,16 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
   }
 
   test("SPARK-24781: Using a reference not in aggregation in Filter/Sort") {
-     withSQLConf(SQLConf.DATAFRAME_RETAIN_GROUP_COLUMNS.key -> "false") {
-      val df = Seq(("test1", 0), ("test2", 1)).toDF("name", "id")
+    spark.sessionState.conf.setConf(SQLConf.DATAFRAME_RETAIN_GROUP_COLUMNS, false)
```

Comment:
nit: how about defining a short name (e.g., `setConf`?) for `spark.sessionState.conf.setConf` in `SharedSparkSessionWithFreshCopy`?  (`spark.sessionState.conf.setConf` is long for typing in...)